### DISCUSSION TITLE
command的同步callback中如何处理异步方法？

### DISCUSSION THREAD

**Discussion by sectlying** [koishijs/koishi#1416] [Q&A] [answered]
I'm new to ts.

![image](https://github.com/koishijs/koishi/assets/77864650/69cccb5a-dbec-40ff-b1bd-567b35cb3568)

如图，getServerStatus()方法返回一个Promise（因为这个方法内部涉及到http请求），我理应在前面加上await获取到Server[]，但action的callback不能返回一个Promise。
请问我如何获取到Server[]，对其处理后返回文本给callback？

**Comment by shigma** [comment, marked as answer]
command 的 action 是异步的，加个 async 即可。

```ts
.action(async () => {
  await getServerStatus()
})
```
